db/document.go

```diff
@@ -881,6 +886,49 @@ func (doc *Document) updateChannelHistory(channelName string, seq uint64, additi
 
 }
 
+func (doc *Document) addToChannelSetHistory(channelName string, historyEntry ChannelSetEntry) {
+	// Before adding the entry we need to verify the number of items in the document channel history so verify
+	// whether we need to prune them.
+	// As we iterate over the existing channels we will keep track of the oldest and second oldest entries along with
+	// their location in the slice. If we need to then prune we can 'merge' the oldest and second oldest and remove
+	// the oldest.
+
+	var oldestEntryStartSeq uint64 = math.MaxUint64
+	var secondOldestEntryStartSeq uint64 = math.MaxUint64
+
+	oldestEntryIdx := -1
+	secondOldestEntryIdx := -1
+
+	entryCount := 0
+
+	for entryIdx, entry := range doc.ChannelSetHistory {
+		if entry.Name == channelName {
+			entryCount++
+
+			if entry.Start < oldestEntryStartSeq {
+				secondOldestEntryStartSeq = oldestEntryStartSeq
+				oldestEntryStartSeq = entry.Start
+
+				secondOldestEntryIdx = oldestEntryIdx
+				oldestEntryIdx = entryIdx
+				continue
+			}
+
+			if entry.Start < secondOldestEntryStartSeq {
+				secondOldestEntryStartSeq = entry.Start
+				secondOldestEntryIdx = entryIdx
+			}
+		}
+	}
+
+	if entryCount == DocumentHistoryMaxEntriesPerChannel {
```

Comment:
It's a bit more defensive here by doing this, just in the event we have some way of adding an entry without pruning.

```suggestion
	if entryCount >= DocumentHistoryMaxEntriesPerChannel {
```

Reply:
Yeah fair enough. Don't think that should happen but this should limit the damage of that possibility